lib/tls/aws_tls.c

```diff
@@ -801,7 +801,7 @@ BaseType_t TLS_Send( void * pvContext,
                 /* Sent data, so update the tally and keep looping. */
                 xWritten += ( size_t ) xResult;
             }
-            else if( 0 == xResult )
+            else if( ( 0 == xResult ) || ( -pdFREERTOS_ERRNO_ENOSPC == xResult ) )
```

Comment:
Since we're not treating the ENOSPC condition as an error here, should we set xResult back to zero?

Reply:
ya that is correct, so that we set the bytes sent as 0.